decayAmplitude/ampIntegralMatrix.cc

```diff
@@ -370,6 +373,41 @@ ampIntegralMatrix::integrate(const vector<const amplitudeMetadata*>& ampMetadata
 		useWeight = true;
 	}
 
+	TTree* eventTree = 0;
+	vector<double> binningVariables(otfBin.size());
+	vector<pair<double, double> > bounds(otfBin.size());
+	if(eventMeta) {
+		if(otfBin.empty()) {
+			printErr << "got event metadata but the binning map is emtpy." << endl;
```

Comment:
Because both `eventMeta` and `otfBin` have default argument, so if one only supplies the first argument of the two and leaves the second on default, this is an error.

A better way to do this might be passing in a `pair<eventMeta*, otfBin>`, would you agree?